projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html

```diff
@@ -28,126 +28,62 @@
-                <mat-expansion-panel-header>
-                    <mat-panel-title fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
-                        <mat-icon>looks</mat-icon>
-                        <span fxFlex>Update Color Map</span>
-                    </mat-panel-title>
-                </mat-expansion-panel-header>
-
-                <ng-template matExpansionPanelContent>
-                    <div class="histogram" *ngIf="histogramCreated === true">
-                        <ng-container *ngIf="histogramData | async as histogramData">
-                            <geoengine-vega-viewer
-                                [chartData]="histogramData"
-                                (interactionChange)="updateBounds($any($event))"
-                            ></geoengine-vega-viewer>
-                        </ng-container>
-                        <mat-progress-bar mode="indeterminate" *ngIf="histogramLoading | async"></mat-progress-bar>
-                    </div>
-                    <div class="histogram-actions">
-                        <button mat-stroked-button class="button" (click)="updateHistogram()">
-                            <ng-container *ngIf="histogramCreated; then createdBlock; else notCreatedBlock"></ng-container>
-                            <!-- Only displayed before any histogram is created -->
-                            <ng-template #createdBlock>Update Histogram</ng-template>
-                            <!-- Only displayed after histogram is created -->
-                            <ng-template #notCreatedBlock>Create Histogram</ng-template>
-                        </button>
-                    </div>
-
-                    <mat-divider></mat-divider>
-
-                    <geoengine-color-map-selector
-                        (breakpointsChange)="updateBreakpoints($event)"
-                        (changesToForm)="getNotified()"
-                        [minValue]="layerMinValue"
-                        [maxValue]="layerMaxValue"
-                        [scale]="scale"
+                <div>
+                    <mat-button-toggle-group
+                        id="colorizer-type"
+                        [multiple]="false"
+                        [ngModel]="getColorizerType()"
+                        (ngModelChange)="updateColorizerType($event)"
                     >
-                    </geoengine-color-map-selector>
-                </ng-template>
-            </mat-expansion-panel>
+                        <mat-button-toggle [value]="linearGradientColorizerType">Linear Gradient</mat-button-toggle>
+                        <mat-button-toggle [value]="logarithmicGradientColorizerType">Logarithmic Gradient</mat-button-toggle>
+                        <mat-button-toggle [value]="paletteColorizerType">Palette</mat-button-toggle>
+                        <mat-button-toggle [value]="rgbaColorizerType">RGBA</mat-button-toggle>
```

Comment:
once you set a (normal) layer from one of the first three to RGB and then go back, you can't set any colors, histogram is broken and backend panics.

Reply:
[81191d2](https://github.com/geo-engine/geoengine-ui/pull/647/commits/81191d2d14407c490b2e202675f199ba405cb9d8)

Yeah, it is missing information… I changed it so that it does not crash.